spec/parcels/parcel-error-handlers.spec.js

```diff
@@ -0,0 +1,169 @@
+import * as singleSpa from 'single-spa'
+
+describe('parcel errors', () => {
+
+  let errs
+
+  function handleError(err) {
+    errs.push(err)
+  }
+
+  beforeAll(() => {
+    singleSpa.start()
+  })
+
+  beforeEach(() => {
+    errs = []
+    singleSpa.addErrorHandler(handleError)
+  })
+
+  afterEach(() => {
+    expect(singleSpa.removeErrorHandler(handleError)).toBe(true)
+  })
+
+  describe('lifecycle errors', () => {
+
+    describe('bootstrap errors', () => {
+
+      it(`should throw an error when mounting fails`, () => {
+        const app = createApp();
+        let shouldAppBeMounted = true;
+
+        singleSpa.registerApplication('parcel-bootstrap-errors', app, () => shouldAppBeMounted);
+        return singleSpa.triggerAppChange().then(() => {
+          expect(app.mountCalls).toBe(1)
+
+          const parcelConfig1 = createParcelConfig('bootstrap')
+          parcelConfig1.name = 'bootstrap-error'
+          const parcel1 = app.mountProps.mountParcel(parcelConfig1, {domElement: document.createElement('div')})
+          return parcel1.bootstrapPromise.then((results) => {
```

Comment:
Shouldn't bootstrapPromise throw if the parcel fails to mount?

Reply:
That's what I expected at first as well, but because we're using the same method as Applications we end up catching the error [here](https://github.com/CanopyTax/single-spa/blob/master/src/applications/lifecycles/bootstrap.js#L15) so in order to actually get the error we have to use an `errorHandler` just like the tests [here](https://github.com/CanopyTax/single-spa/blob/4.0/spec/apis/error-handlers.spec.js#L37)